android/src/main/kotlin/com/example/android_locator_plugin/AndroidLocatorPlugin.kt

```diff
@@ -1,35 +1,180 @@
+            locationRequest.interval = locationRequestInterval
+            locationRequest.priority = PRIORITY_HIGH_ACCURACY
+            locationCallback = object : LocationCallback() {
+                override fun onLocationResult(locationResult: LocationResult?) {
+                    locationEventSource?.success(geographicLongitude + locationResult?.lastLocation?.longitude.toString() + " - " + geographicLatitude + locationResult?.lastLocation?.latitude.toString())
+                }
+            }
+        } else {
+            locationEventSource?.success(accessDenied)
+        }
+        result.success(permission)
+    }
+
+    private fun permissionGranted(): Boolean {
+        var granted: Boolean
+        context.run {
+            granted = ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+        }
+        return granted
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activity = binding.activity
+        binding.addActivityResultListener(this)
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        Log.i("onDetachedFromActivity", "No longer valid")
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        Log.i("onReattachedToActivity", "No longer valid")
+    }
+
+    override fun onDetachedFromActivity() {
+        Log.i("onDetachedFromActivity", "No longer valid")
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
+        Log.i("onActivityResult", "No longer valid")
+        return true
+    }
+
+    companion object PluginConstants {
+        const val initialize: String = "initialize"
+        const val locationRequestInterval: Long = 1500
+        const val requestCode: Int = 500
+        const val geographicLongitude: String = "Longitude: "
+        const val geographicLatitude: String = "Latitude: "
+        const val accessGranted: String = "Your access is granted!"
+        const val accessDenied: String = "Your access is not granted"
+        const val checkPermission: String = "checkPermission"
+        const val accessStream: String = "access_event_channel"
+        const val eventLocation: String = "location_event_channel"
+        const val androidLocatorPlugin: String = "android_locator_plugin"
+        const val requestPermission: String = "requestPermission"
+        const val pluginInitialized: String = "The plugin has been initialized"
+        const val pluginNeedPermission: String = "You need access first"
```

Comment:
Variables on android are defined with uppercase separated each word with underscore. So instead to "initialize" you can use INITIALIZE, and is not necessary to define the type so define a variable like this: 
const val INITIALIZE = "initialize" / const val ACCESS_STREAM = "access_event_channel"

Reply:
Thanks for the lesson. Variables are now defined in uppercase